opentelemetry-user-events-logs/src/logs/exporter.rs

```diff
@@ -11,20 +11,64 @@ use std::{cell::RefCell, str, time::SystemTime};
 
 thread_local! { static EBW: RefCell<EventBuilder> = RefCell::new(EventBuilder::new());}
 
-/// UserEventsExporter is a log exporter that exports logs in EventHeader format to user_events tracepoint.
-pub(crate) struct UserEventsExporter {
-    provider: Mutex<Provider>,
-    name: String,
-    event_sets: Vec<Arc<EventSet>>,
-    cloud_role: Option<String>,
-    cloud_role_instance: Option<String>,
+/// Options for configuring the User Events Exporter.
+#[derive(Debug)]
+pub struct ExportOptions<'a> {
+    provider_name: &'a str,
 }
 
-const EVENT_ID: &str = "event_id";
+impl<'a> ExportOptions<'a> {
+    /// Returns the provider name.
+    fn provider_name(&self) -> &'a str {
+        self.provider_name
+    }
+
+    /// Builder for `EventsOptions`, with a mandatory provider_name.
+    /// This is the name of the provider that will be used to register the event sets.
+    /// The provider name must:
+    /// - Not be empty.
+    /// - Be less than 234 characters.
+    /// - Contain only ASCII letters, digits, and the underscore (`'_'`) character.
+    /// - Be short, human-readable, and unique enough to avoid conflicts with other provider names.
+    /// - Typically include a company name and a component name, e.g., "MyCompany_MyComponent".
+    ///
+    /// If an invalid provider name is provided, this method will not add the exporter to the builder.
+    ///
+    /// Tracepoint names are generated by combining the provider name, event
+    /// level and keyword (currently hardcoded to `1`) in the following format:
+    /// `ProviderName + '_' + 'L' + EventLevel + 'K' + EventKeyword`
+    ///
+    /// For example, if "myprovider" is the provider name, the following tracepoint names are created:
+    /// - `myprovider_L5K1`
+    /// - `myprovider_L4K1`
+    /// - `myprovider_L3K1`
+    /// - `myprovider_L2K1`
+    /// - `myprovider_L1K1`
+    ///
+    /// perf tool can be used to record events from the tracepoints.
+    /// For example the following will capture level 2 (Error) and 3(Warning) events:
+    /// perf record -e user_events:myprovider_L2K1,user_events:myprovider_L3K1
+    pub fn builder(provider_name: &'a str) -> ExportOptionsBuilder<'a> {
```

Comment:
```suggestion
    pub fn builder(provider_name: impl AsRef<Str>) -> ExportOptionsBuilder<'a> {
```

Reply:
@utpilla - this needs to be actually changed to:
```rust
    pub fn builder<S: AsRef<str> + ?Sized>(provider_name: &'a S) -> ExportOptionsBuilder<'a> {
        ExportOptionsBuilder::new(provider_name)
    }
```
for both ExportOptionsBuilder and ExportOptions.

This is doable, but I feel 
```rust
pub fn builder(provider_name: &'a str) -> ExportOptionsBuilder<'a>
```
is more simpler. Unless we really need the added flexibility coming from `&'a S: AsRef<str>`.